RecoJets/JetProducers/src/JetIDHelper.cc

```diff
@@ -341,15 +347,17 @@ void reco::helper::JetIDHelper::classifyJetComponents( const edm::Event& event,
 	  hitE = theRecHit->energy();
 	  int iEta = theRecHit->id().ieta();
 	  int depth = theRecHit->id().depth();
-	  Region region = HBHE_region( iEta, depth );
+	  Region region = HBHE_region( theRecHit->id().rawId() );
 	  int hitIPhi = theRecHit->id().iphi();
 	  if( iDbg>3 ) cout<<"hit #"<<iCell<<" is HBHE, E: "<<hitE<<" iEta: "<<iEta
 			   <<", depth: "<<depth<<", iPhi: "<<theRecHit->id().iphi()
 			   <<" -> "<<region;
+
+	  std::vector<int> isMergedDepth = theHcalTopology->mergedDepthList29(theRecHit->id());
 	  int absIEta = TMath::Abs( theRecHit->id().ieta() );
-	  if( depth == 3 && (absIEta == 28 || absIEta == 29) ) {
-	    hitE /= 2; // Depth 3 at the HE forward edge is split over tower 28 & 29, and jet reco. assigns half each
-	  }
+	  if( (absIEta == 28 || absIEta == 29) &&  std::find(isMergedDepth.begin(), isMergedDepth.end(), depth) != isMergedDepth.end() )
```

Comment:
given this usage of the vector from mergedDepthList29, it would be better just to have a 

bool isMergedDepth = theHcalTopology->mergedDepthList29(theRecHit->id())